File: Frontend/src/types/link.ts
/* Link-related types. */

export interface Link {
  id: string;
  tenant_id: string;
  title: string;
  url: string;
  icon: string | null;
  position: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface LinkCreateRequest {
  title: string;
  url: string;
  icon?: string;
}

export interface LinkUpdateRequest {
  title?: string;
  url?: string;
  icon?: string;
  is_active?: boolean;
}

export interface LinkOrderItem {
  id: string;
  position: number;
}

export interface LinkReorderRequest {
  order: LinkOrderItem[];
}

File: Backend/app/schemas/link.py
"""Link schemas."""

from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional


class LinkResponse(BaseModel):
    id: UUID
    tenant_id: UUID
    title: str
    url: str
    icon: Optional[str] = None
    position: int
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LinkCreateRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    url: str = Field(..., min_length=1, max_length=2048)
    icon: Optional[str] = Field(None, max_length=100)


class LinkUpdateRequest(BaseModel):
    title: Optional[str] = Field(None, max_length=255)
    url: Optional[str] = Field(None, max_length=2048)
    icon: Optional[str] = Field(None, max_length=100)
    is_active: Optional[bool] = None


class LinkOrderItem(BaseModel):
    id: UUID
    position: int


class LinkReorderRequest(BaseModel):
    order: list[LinkOrderItem]
